sys/net/routing/nhdp/iib_table.c

```diff
@@ -0,0 +1,696 @@
+
+/* Internal function prototypes */
+static void rem_link_set_entry(iib_base_entry_t *base_entry, iib_link_set_entry_t *ls_entry);
+static void cleanup_link_sets(nhdp_addr_entry_t *rem_list);
+static iib_link_set_entry_t* add_default_link_set_entry(iib_base_entry_t *base_entry, timex_t *now,
+        uint64_t val_time);
+static void reset_link_set_entry(iib_link_set_entry_t *ls_entry, timex_t *now, uint64_t val_time);
+static iib_link_set_entry_t* update_link_set(iib_base_entry_t *base_entry, nib_entry_t *nb_elt,
+        nhdp_addr_entry_t *send_list, timex_t *now, uint64_t val_time, uint8_t sym, uint8_t lost);
+static void release_link_tuple_addresses(iib_link_set_entry_t *ls_entry);
+
+static int update_two_hop_set(iib_base_entry_t *base_entry, iib_link_set_entry_t *ls_entry,
+        nhdp_addr_entry_t *th_sym_list, nhdp_addr_entry_t *th_rem_list,
+        timex_t *now, uint64_t val_time);
+static uint8_t rem_exst_th_entries(iib_base_entry_t *base_entry,
+        iib_two_hop_set_entry_t *ts_elt, nhdp_addr_entry_t *th_sym_list);
+static uint8_t rem_non_sym_th_entries(iib_base_entry_t *base_entry,
+        iib_two_hop_set_entry_t *ts_elt, nhdp_addr_entry_t *th_rem_list);
+static int add_two_hop_entry(iib_base_entry_t *base_entry, iib_link_set_entry_t *ls_entry,
+        nhdp_addr_t *th_addr, timex_t *now, uint64_t val_time);
+static void rem_two_hop_entry(iib_base_entry_t *base_entry, iib_two_hop_set_entry_t *th_entry);
+
+static void wr_update_ls_status(iib_base_entry_t *base_entry,
+        iib_link_set_entry_t *ls_elt, timex_t *now);
+static void update_nb_tuple_symmetry(iib_base_entry_t *base_entry,
+        iib_link_set_entry_t *ls_entry, timex_t *now);
+static void rem_not_heard_nb_tuple(iib_link_set_entry_t *ls_entry, timex_t *now);
+
+static inline timex_t get_max_timex(timex_t time_one, timex_t time_two);
+static iib_link_tuple_status_t get_tuple_status(iib_link_set_entry_t *ls_entry, timex_t *now);
+
+
+/*---------------------------------------------------------------------------*
+ *                       Interface Information Base API                      *
+ *---------------------------------------------------------------------------*/
+
+int iib_register_if(kernel_pid_t pid)
+{
+    iib_base_entry_t *new_entry = (iib_base_entry_t*) malloc(sizeof(iib_base_entry_t));
+    if (!new_entry) {
+        /* Insufficient memory */
+        return -1;
+    }
+    new_entry->if_pid = pid;
+    new_entry->link_set_head = NULL;
+    new_entry->two_hop_set_head = NULL;
+    LL_PREPEND(iib_base_entry_head, new_entry);
+
+    return 0;
+}
+
+int iib_process_hello(kernel_pid_t if_pid, nib_entry_t *nb_elt, nhdp_addr_entry_t *send_list,
+        nhdp_addr_entry_t *th_sym_list, nhdp_addr_entry_t *th_rem_list,
+        nhdp_addr_entry_t *rem_list, uint64_t validity_time, uint8_t is_sym_nb, uint8_t is_lost)
+{
+    iib_base_entry_t *base_elt;
+    timex_t now;
+
+    mutex_lock(&mtx_iib_access);
+
+    vtimer_now(&now);
```

Comment:
Would it make sense to move  this inside the `ìf(base_elt)` block? (To save the call in case no element is found for the interface and to increase accuracy of the value.)